### PR TITLE
add KEP for vpa manager feature doc

### PR DESCRIPTION
**Pull request by EchoUtopia**
#### Which component this PR applies to?

vertical-pod-autoscaler

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
as worklods like deployments,statefulsets,etc. numbers growing, managing vpas become more and more complicated, because we need to manually create vpa for each of them. in more common scenarios, we just need to to create vpa for majority of workloads, and let vpa to extend or shrink workloads resources quietly. with vpa manager, we can just create or delete vpa in the scope of namespace, vpa manager will automaticlly create or delete for all workloads in the namespace.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

NONE
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mwielgus** [comment]
@jbartosik Please take a look.

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+we want to make vpa more easier and automatically to use.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- use labels autoscaling/vpa=open or autoscaling/vpa=open to create or delete vpa or set some parameters for workloads in namespaces
```
Comment (jbartosik) [inline vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:166]: So we'd have:
- label on namespace that tells VPA manager that it should create VPA objects for workloads in that namespace
- label on namespace that tells VPA manager what mode workloads in the namespace should have
- label on workload that tells VPA manager that it shouldn't create VPA object for the workload.

Did I understand that correctly?

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+  when we want to create vpa for all workloads in namespace a, we just label namespace a with autoscaling/vpa=open, then vpa manager create vpa for every workload not annotated with autoscaling/vpa=close in namespace a, and later when workload are created in namespace a, corresponding vpa will be automaticlly created.
```
Comment (jbartosik) [inline vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:200]: In this story there is no `vpa/update-mode` label. What mode do VPA objects have?

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+  when we want to delete all vpas in namespace a, we just label namespace a with autoscaling/vpa=close, then vpa manager delete vpa for every workload not annotated with autoscaling/vpa=open in namespace a.
+
+#### Story 2
+
+  we can configure vpa manger default pod update mode as auto( or initial). when vpa manager create vpa, it will set vpa's updatepolicy.updateMode to auto (or initial). we can also specify update mode for namespace a by labeling namespace a with vpa/update-mode=initial( or vpa/update-mode=initial), in this case, all vpa created in namespace a has initial update mode except those vpa correspoding workloads annotated vpa/update-mode=auto.
+
+  later when we change update mode of namespace a by labeling it, all the vpa's update mode will be changed correspondingly in namespace a except those tarRef has been annotated by vpa/update-mode=***.
```
Comment (jbartosik) [inline vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:208]: So this means I can't change VPA mode by editing VPA, I need to change mode by adding a label, right?

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+later i will create a pr
```
Comment (jbartosik) [inline vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:241]: I think we need more details here.

Is it a separate executable? What happens with unusual label configurations? Why are we using labels as an API? What will the updater do (presumably watch namespaces, watch all controllers, watch VPA objects, do something to ensure VPA match labels on namespaces and controllers)

**Review by jbartosik — CHANGES_REQUESTED** [review]
I left a few small comments and 2 I think are more important:

- please remove comments to make the KEP easier to read
- please add some more details in the design section

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
+<!--
+**Note:** When your KEP is complete, all of these comment blocks should be removed.
```
[the remark this answers is not in the document]
Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:2]: thank you ,i'll improve it soon

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:166]: sorry, it's autoscaling/vpa=open or autoscaling/vpa=close. my English is poor and i'll check more carefully

Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:166]: yes, exactly

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md (hunk 2 above)
Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:200]: thre's a default update mode configured in vpa manager launch parameters

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md (hunk 3 above)
Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:208]: yes, it's for common use, if anyone want custom vpa parameters, he can annotate workloads with label "autoscaling/vpa=close" and create vpa manually

File: vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md (hunk 4 above)
Comment (EchoUtopia) [reply vertical-pod-autoscaler/enhancements/5030-vpa-manager/README.md:241]: it's a seperate executable

**Pull request closed without merging**